python_modules/dagster/dagster/_core/definitions/load_assets_from_modules.py

```diff
@@ -414,6 +420,16 @@ def replace_keys_in_asset(
         if isinstance(asset, AssetsDefinition)
         else asset.with_attributes(key=key_replacements.get(asset.key, asset.key))
     )
+    if isinstance(asset, AssetChecksDefinition):
+        updated_object = cast(AssetsDefinition, updated_object)
```

Comment:
I think it's only necessary because of test strictness. i think we can loosen and change tests but feels like a followup. Keeping tests unchanged seems like a plus for a refactor like this